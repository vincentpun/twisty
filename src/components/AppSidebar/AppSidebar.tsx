import * as React from 'react';

import HeaderBar from './components/HeaderBar';

interface AppSidebarProps {
  children?: JSX.Element;
  productName: string;
}

const AppSidebar = ({ children, productName }: AppSidebarProps) => (
  <div className="app-sidebar">
    <HeaderBar className="app-sidebar__header-bar" title={productName} />
    {children}
  </div>
)

export default AppSidebar;
