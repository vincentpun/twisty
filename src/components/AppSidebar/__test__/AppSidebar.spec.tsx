import * as React from 'react';
import { shallow } from 'enzyme';

import AppSidebar from '../AppSidebar';

describe('AppSidebar', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<AppSidebar productName="" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders children', () => {
    const wrapper = shallow(
      <AppSidebar productName="">
        <div className="test">Test</div>
      </AppSidebar>
    );

    expect(wrapper.find('.test').length).toBe(1);
    expect(wrapper.find('.test').first().text()).toBe('Test');
  });
});
