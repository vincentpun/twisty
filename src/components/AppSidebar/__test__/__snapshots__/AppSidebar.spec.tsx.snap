// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSidebar matches the snapshot 1`] = `
<div
  className="app-sidebar"
>
  <HeaderBar
    className="app-sidebar__header-bar"
    title=""
  />
</div>
`;
